src/index.ts

```diff
@@ -51,13 +51,15 @@ export class StreamLogger {
 	 *	standalone for special situations.
 	 */
 	public formatWithPrefix(name: string, message: string): string {
-		const maxLength = this.getMaxPrefixLength()
 		const prefix = this.prefixes[name]
+		if (!prefix) throw new Error(`Unknown prefix ${name}`)
+
+		const maxLength = this.getMaxPrefixLength()!
 		const spaces = _.times(maxLength - prefix.length, _.constant(' ')).join('')
 		return `${prefix}${spaces} ${message}\n`
 	}
 
-	private getMaxPrefixLength(): number {
+	private getMaxPrefixLength(): number | undefined {
```

Comment:
Forgive me for wading in on code I've never even looked at before, but I guess this max-prefix-length could be stored as a class-attribute (i.e. an additional property on `this`), as it will only need to get updated in the constructor and when `addPrefix` is called, and doesn't need to be re-calculated _every_ time that `formatWithPrefix` is called ?

Reply:
It _could_, yes. It's more complicated and a little more error-prone though (e.g. if we add more methods changing the prefixes available, we need to remember to handle this). I'd definitely do that if this was an issue, but right now I'd be fairly surprised if calling `getMaxPrefixLength` unnecessarily had a measurable effect on performance in any real situation.

So yes, in principle, but until we see a concrete performance cost here I plan to wholeheartedly ignore this otherwise sensible suggestion :smile:.